Publishing the devcontainer fails because the image does not exist locally

```console
Run set -e
  set -e
  
  # Push the image to Docker Hub
  docker push "sagebionetworks/challenge-devcontainer:4287403b"
  shell: /usr/bin/bash -e {0}
  env:
    IMAGE_REPOSITORY: sagebionetworks/challenge-devcontainer
    DEFAULT_BRANCH: main
The push refers to repository [docker.io/sagebionetworks/challenge-devcontainer]
An image does not exist locally with the tag: sagebionetworks/challenge-devcontainer
Error: Process completed with exit code 1.
```

--- a/.github/workflows/build-devcontainer-image.yml
+++ b/.github/workflows/build-devcontainer-image.yml
@@ -45,7 +45,7 @@ jobs:
           set -e
 
           DEVCONTAINER_VERSION="${GITHUB_SHA::8}"
-          IMAGE_NAME="${{ env.IMAGE_REPOSITORY }}:${DEVCONTAINER_VERSION}"
+          IMAGE_NAME="${{ env.IMAGE_REPOSITORY }}"
           echo ::set-output name=image_name::${IMAGE_NAME}
           echo ::set-output name=devcontainer_version::${DEVCONTAINER_VERSION}
 
@@ -55,18 +55,26 @@ jobs:
 
           FOLDER_WITH_DOT_DEVCONTAINER="tools/devcontainers/challenge-devcontainer"
           IMAGE_NAME=${{ steps.prep.outputs.image_name }}
+          IMAGE_TAG=${{ steps.prep.outputs.devcontainer_version }}
 
           # [Optional] Enable buildkit, set output to plain text for logging
           export DOCKER_BUILDKIT=1
           export BUILDKIT_PROGRESS=plain
 
-          # Export the devcontainer version, which will be set inside the image. See 
+          # Export the devcontainer version, which will be set inside the image. See
           # devcontainer.json used to build the image.
           export DEVCONTAINER_VERSION=${{ steps.prep.outputs.devcontainer_version }}
 
           # Build the image
-          npm install -g "@vscode/dev-container-cli"
-          devcontainer build --image-name "${IMAGE_NAME}" "${FOLDER_WITH_DOT_DEVCONTAINER}"
+          npm install -g @vscode/dev-container-cli
+          devcontainer build \
+            --disable-telemetry true \
+            --image-name "${IMAGE_NAME}:${IMAGE_TAG}" "${FOLDER_WITH_DOT_DEVCONTAINER}"
+
+          # Check that the image exists because `devcontainer` seems to always returns 0
+          if [[ "$(docker images -q "${IMAGE_NAME}:${IMAGE_TAG}" 2> /dev/null)" == "" ]]; then
+            exit 1
+          fi
 
       - name: Login to Docker Hub
         if: ${{ github.event_name != 'pull_request' }}
@@ -78,7 +86,6 @@ jobs:
       - name: Push the image to Docker Hub
         if: ${{ github.event_name != 'pull_request' }}
         run: |
-          set -e
-
-          # Push the image to Docker Hub
-          docker push "${{ steps.prep.outputs.image_name }}"
+          IMAGE_NAME=${{ steps.prep.outputs.image_name }}
+          IMAGE_TAG=${{ steps.prep.outputs.devcontainer_version }}
+          docker push "${IMAGE_NAME}:${IMAGE_TAG}"

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -35,9 +35,9 @@ RUN groupadd docker \
     nodejs gh ngrok \
   && npm install --global yarn@latest \
   # Install nektos/act
-  && curl https://raw.githubusercontent.com/nektos/act/v${actVersion}/install.sh | bash - \
+  && curl "https://raw.githubusercontent.com/nektos/act/v${actVersion}/install.sh" | bash - \
   # Install hyperfine
-  && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/v${hyperfineVersion}/hyperfine_${hyperfineVersion}_amd64.deb \
+  && curl -sSL "https://github.com/sharkdp/hyperfine/releases/download/v${hyperfineVersion}/hyperfine_${hyperfineVersion}_amd64.deb" \
     -o /tmp/hyperfine.deb \
   && apt-get install --no-install-recommends -qq -y /tmp/hyperfine.deb \
   && rm -fr /tmp/hyperfine.deb \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -19,7 +19,7 @@ RUN groupadd docker \
   # Add Node.js repository
   && curl -sL https://deb.nodesource.com/setup_16.x | bash - \
   # Add GitHub CLI repository
-  && curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | \
+  && curl -sSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | \
       gpg --dearmor -o /usr/share/keyrings/githubcli-archive-keyring.gpg \
   && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | \
       tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
@@ -34,9 +34,11 @@ RUN groupadd docker \
   # Install nektos/act
   && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
   # Install hyperfine
-  && curl https://github.com/sharkdp/hyperfine/releases/download/v1.14.0/hyperfine_1.14.0_amd64.deb \
-  && apt-get install --no-install-recommends -qq -y ./hyperfine_1.14.0_amd64.deb \
-  && rm -fr hyperfine_1.14.0_amd64.deb \
+  && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/v1.14.0/hyperfine_1.14.0_amd64.deb \
+    -o /tmp/hyperfine.deb \
+  && apt-get install --no-install-recommends -qq -y /tmp/hyperfine.deb \
+  && rm -fr /tmp/hyperfine.deb \
+  # Cleanup
   && apt-get -y autoclean \
   && apt-get -y autoremove \
   && rm -rf /var/lib/apt/lists/* \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -7,7 +7,7 @@ SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
 ARG devcontainerVersion=""
 ENV DEVCONTAINER_VERSION=${devcontainerVersion}
 
-ARG hyperfineVersion="v1.14.0"
+ARG hyperfineVersion="1.14.0"
 
 # Create the docker group so that we can assign it to the user.
 # This is to enable the non-root user to use the command `docker`.
@@ -36,7 +36,7 @@ RUN groupadd docker \
   # Install nektos/act
   && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
   # Install hyperfine
-  && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/${hyperfineVersion}/hyperfine_1.14.0_amd64.deb \
+  && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/v${hyperfineVersion}/hyperfine_${hyperfineVersion}_amd64.deb \
     -o /tmp/hyperfine.deb \
   && apt-get install --no-install-recommends -qq -y /tmp/hyperfine.deb \
   && rm -fr /tmp/hyperfine.deb \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -7,8 +7,8 @@ SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
 ARG devcontainerVersion=""
 ENV DEVCONTAINER_VERSION=${devcontainerVersion}
 
-ARG hyperfineVersion="1.14.0" \
-  actVersion="0.2.30"
+ARG actVersion="0.2.30"
+ARG hyperfineVersion="1.14.0"
 
 # Create the docker group so that we can assign it to the user.
 # This is to enable the non-root user to use the command `docker`.

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -35,7 +35,7 @@ RUN groupadd docker \
   && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
   # Install hyperfine
   && curl https://github.com/sharkdp/hyperfine/releases/download/v1.14.0/hyperfine_1.14.0_amd64.deb \
-  && apt install ./hyperfine_1.14.0_amd64.deb \
+  && apt-get install ./hyperfine_1.14.0_amd64.deb \
   && rm -fr hyperfine_1.14.0_amd64.deb \
   && apt-get -y autoclean \
   && apt-get -y autoremove \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -7,7 +7,8 @@ SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
 ARG devcontainerVersion=""
 ENV DEVCONTAINER_VERSION=${devcontainerVersion}
 
-ARG hyperfineVersion="1.14.0"
+ARG hyperfineVersion="1.14.0" \
+  actVersion="0.2.30"
 
 # Create the docker group so that we can assign it to the user.
 # This is to enable the non-root user to use the command `docker`.
@@ -34,7 +35,7 @@ RUN groupadd docker \
     nodejs gh ngrok \
   && npm install --global yarn@latest \
   # Install nektos/act
-  && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
+  && curl https://raw.githubusercontent.com/nektos/act/v${actVersion}/install.sh | bash - \
   # Install hyperfine
   && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/v${hyperfineVersion}/hyperfine_${hyperfineVersion}_amd64.deb \
     -o /tmp/hyperfine.deb \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -1,11 +1,13 @@
 ARG imageVersion=bullseye-20211220-slim
 FROM debian:$imageVersion
 
+# Safer bash scripts with 'set -euxo pipefail'
+SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
+
 ARG devcontainerVersion=""
 ENV DEVCONTAINER_VERSION=${devcontainerVersion}
 
-# Safer bash scripts with 'set -euxo pipefail'
-SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
+ARG hyperfineVersion="v1.14.0"
 
 # Create the docker group so that we can assign it to the user.
 # This is to enable the non-root user to use the command `docker`.
@@ -34,7 +36,7 @@ RUN groupadd docker \
   # Install nektos/act
   && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
   # Install hyperfine
-  && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/v1.14.0/hyperfine_1.14.0_amd64.deb \
+  && curl -sSL https://github.com/sharkdp/hyperfine/releases/download/${hyperfineVersion}/hyperfine_1.14.0_amd64.deb \
     -o /tmp/hyperfine.deb \
   && apt-get install --no-install-recommends -qq -y /tmp/hyperfine.deb \
   && rm -fr /tmp/hyperfine.deb \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -35,7 +35,7 @@ RUN groupadd docker \
   && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
   # Install hyperfine
   && curl https://github.com/sharkdp/hyperfine/releases/download/v1.14.0/hyperfine_1.14.0_amd64.deb \
-  && apt-get install ./hyperfine_1.14.0_amd64.deb \
+  && apt-get install --no-install-recommends -qq -y ./hyperfine_1.14.0_amd64.deb \
   && rm -fr hyperfine_1.14.0_amd64.deb \
   && apt-get -y autoclean \
   && apt-get -y autoremove \

--- a/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
+++ b/tools/devcontainers/challenge-devcontainer/.devcontainer/Dockerfile
@@ -35,7 +35,7 @@ RUN groupadd docker \
   && curl https://raw.githubusercontent.com/nektos/act/v0.2.30/install.sh | bash - \
   # Install hyperfine
   && curl https://github.com/sharkdp/hyperfine/releases/download/v1.14.0/hyperfine_1.14.0_amd64.deb \
-  && dpkg -i hyperfine_1.14.0_amd64.deb \
+  && apt install ./hyperfine_1.14.0_amd64.deb \
   && rm -fr hyperfine_1.14.0_amd64.deb \
   && apt-get -y autoclean \
   && apt-get -y autoremove \

--- a/tools/devcontainers/challenge-devcontainer/build-image.sh
+++ b/tools/devcontainers/challenge-devcontainer/build-image.sh
@@ -4,4 +4,5 @@
 
 devcontainer build \
   --no-cache \
+  --disable-telemetry true \
   --image-name sagebionetworks/challenge-devcontainer:test

